### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by GavenElder**
added feature, tests, comments. 
FINAL SUBMIT

### CODE REVIEW

**Comment by A-Tanner** [comment]
I will review this

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 1]
```diff
@@ -90,7 +98,7 @@ public static Question[] LoadQuestions(string filePath)
                 question.Answers[2] = answer3;
                 question.CorrectAnswerIndex = correctAnswerIndex;
             }
-            return questions;
+            return RandomQuestionGenerator(questions);
```
Comment (A-Tanner) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:101]: RandomQuestionGenerator is given an array of null questions, so the main throws a NullReferenceException when it goes to read the questions

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 2]
```diff
@@ -65,6 +68,11 @@ public static string GetFilePath()
         {
             return "Trivia.txt";
         }
+        public static Question[] RandomQuestionGenerator(Question[] questions) // added feature
```
Comment (A-Tanner) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:71]: Style: Misleading function name; this function does not generate anything, but rather, randomizes an existing array.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 3]
```diff
@@ -86,5 +86,15 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
         }
+        [TestMethod]
+        public void RandomizerQuestionGenerator() //tests for added random feature 
+        {
+            string file = Path.GetRandomFileName(); //creates file path
+            GenerateQuestionsFile(file, 2); //creates path for two questions.
+            Question[] firstQuestion = Program.LoadQuestions(file); //loads first question
+            Question[] secondQuestion = Program.LoadQuestions(file);//loads second question
+            File.Delete(file); //deletes file path
+
```
Comment (A-Tanner) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:97]: RandomizerQuestionGenerator test has no fail condition, and therefore cannot prove that the method it is testing has functional logic. Use Assert statements to generate failure conditions, and add the code to test what RandomQuestionGenerator should be doing

**Review by A-Tanner — CHANGES_REQUESTED** [review]
Assignment
Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ❌ -->NullReferenceException on question generation
Issue 2: Unit test properly passes ✔
Extra Credit
Issue 3: Feature request implemented ❌ --> Although the present logic should work, the application crashes and therefore the feature is still unimplemented
Issue 3: Feature request unit tested ❌ -->Test has no fail conditions, does not test functionality


Feel free to email (atanner5@ewu.edu) or message me on Discord  (Ray Tanner#2916) with any questions

**Comment by GavenElder** [comment]
Thank you for your PR!

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 4]
```diff
@@ -86,5 +85,24 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
+                //act
+                Question[] firstQuestion = Program.LoadQuestions(filePath);
+                Question[] secondQuestion = Program.LoadQuestions(filePath);
+                bool arrayisequal = firstQuestion.SequenceEqual(secondQuestion);
+                //asset
+                Assert.IsFalse(arrayisequal);
+            }
+            finally
+                File.Delete(filePath);
+
```
Comment (mmwoodfo) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:105]:
```suggestion
            finally
            {
                File.Delete(filePath);
            }

```

This is causing errors in your code.

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 5]
```diff
@@ -86,5 +85,24 @@ private static void GenerateQuestionsFile(string filePath, int numberOfQuestions
...
+            try
+            {
+                //arrange
+                GenerateQuestionsFile(filePath, 2);
+                //act
+                Question[] firstQuestion = Program.LoadQuestions(filePath);
+                Question[] secondQuestion = Program.LoadQuestions(filePath);
+                bool arrayisequal = firstQuestion.SequenceEqual(secondQuestion);
+                //asset
+                Assert.IsFalse(arrayisequal);
```
Comment (mmwoodfo) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:101]: You can also use `CollectionAssert.AreEqual` here for your arrays. But this solution is also acceptable.

Something to consider, there is a chance that the two arrays are shuffled in the same order. A way around this is to do a few shuffles and make sure that at least two of the indexes differ.

**Comment by mmwoodfo** [comment]
@GavenElder  did you do a PR review? If so, can you provide me with the name of the student you did the review for and/or the PR #?

Thank you.

**Comment by GavenElder** [comment]
Yes I did do a PR. I believe it was an older gentlemen with the last name Scott, but I'm not seeing him within the forked repositories anymore???

**Comment by GavenElder** [comment]
It difficult to to see on the Github app on my phone. But I'll find it when I get home.

**Comment by mmwoodfo** [comment]
@GavenElder No worries, I found it. https://github.com/IntelliTect-Samples/EWU-CSCD371-2022-Winter/pull/19
Sorry about that, they can be difficult to track down sometimes.

**Pull request closed without merging**